botocore/loaders.py

```diff
@@ -103,21 +103,34 @@
 """
 import logging
 import os
+import pathlib
+import sys
+from zipfile import is_zipfile
+
+# python < 3.9 botocore will not support zips
+if sys.version_info < (3, 9):
+
+    class ZipPath:
+        def __init__(self):
+            raise RuntimeError(
+                "Using botocore in a zipped context "
+                "is not supported in python < 3.9"
+            )
+
+else:
+    from zipfile import Path as ZipPath
```

Comment:
Since botocore will not support zips in python < 3.9, but `zipfile.Path` objects are native as of python 3.8, this check needs to be performed. I'm not a huge fan of branching logic on imports, but I'm not sure of how else to do this. If anyone knows of a more elegant way to check, let me know.